app/src/main/java/org/wikipedia/page/references/ReferenceDialog.kt

```diff
@@ -91,10 +93,25 @@ class ReferenceDialog : ExtendedBottomSheetDialogFragment() {
             binding.referenceText.movementMethod = LinkMovementMethodExt(callback()?.linkHandler)
         }
 
-        fun bindItem(idText: CharSequence?, contents: CharSequence?) {
+        fun bindItem(idText: CharSequence?, reference: PageReferences.Reference) {
             binding.referenceId.text = idText
             binding.root.post {
                 if (isAdded) {
+                    val contents = StringUtil.fromHtml(StringUtil.removeCiteMarkup(StringUtil.removeStyleTags(reference.html)))
+                    if (contents.isNullOrEmpty()) {
+                        // Inspect html for external links
+                        val links = StringUtil.extractHrefLinks(reference.html)
+                        links.add(links[0])
```

Comment:
Not sure I understand this statement. We're adding the first element of the list to itself? Wouldn't this create an unnecessary duplicate?

Reply:
Was testing multiple links 😅